### PR TITLE
Implement "fetchAll" pagination

### PR DESCRIPTION
**Pull request by znarf**

### CODE REVIEW

File: src/server/lib/utils.js [hunk 1]
```diff
@@ -20,20 +21,9 @@ export const days = (d1, d2 = new Date()) => {
...
-          else if (td !== undefined && td !== null) return td;
-          else return '';
-        })
-        .join('","')}"`,
-    );
-  });
-  return lines.join('\n');
+export function json2csv(data, opts) {
+  const parser = new Parser(opts);
+  return parser.parse(data);
```
Comment (znarf) [inline src/server/lib/utils.js:26]: Replace with a mature library. Ability to skip headers when needed.

File: src/server/controllers/account-transactions.js [hunk 2]
```diff
@@ -256,9 +256,8 @@ const accountTransactions = async (req, res) => {
     'dateTo',
     'minAmount',
     'maxAmount',
-    'flattenHostFee',
   ]);
-  variables.limit = Number(variables.limit) || 1000;
+  variables.limit = variables.limit ? Number(variables.limit) : null;
```
Comment (znarf) [inline src/server/controllers/account-transactions.js:260]: No default limit here, will fallback on the default API side.

File: src/server/controllers/account-transactions.js [hunk 3]
```diff
@@ -293,7 +292,7 @@ const accountTransactions = async (req, res) => {
     variables.kind = variables.kind.split(',').map(toUpper).map(trim);
   }
 
-  variables.fetchHostFee = parseToBooleanDefaultFalse(variables.flattenHostFee);
+  variables.fetchHostFee = parseToBooleanDefaultFalse(req.query.flattenHostFee);
```
Comment (znarf) [inline src/server/controllers/account-transactions.js:295]: `flattenHostFee` is not in variables anymore.

File: src/server/controllers/account-transactions.js [hunk 4]
```diff
@@ -351,14 +352,29 @@ const accountTransactions = async (req, res) => {
         const mapping = pick(csvMapping, fields);
 
         const mappedTransactions = result.transactions.nodes.map((t) => applyMapping(mapping, t));
-
-        let csv = json2csv(mappedTransactions);
+        res.write(json2csv(mappedTransactions));
+        res.write(`\n`);
```
Comment (znarf) [inline src/server/controllers/account-transactions.js:356]: write but don't end the request.

File: src/server/controllers/account-transactions.js [hunk 5]
```diff
@@ -351,14 +352,29 @@ const accountTransactions = async (req, res) => {
...
+          if (fetchAll) {
+            do {
+              variables.offset += result.transactions.limit;
+
+              result = await graphqlRequest(query, variables, { version: 'v2', headers });
+
+              const mappedTransactions = result.transactions.nodes.map((t) => applyMapping(mapping, t));
+              res.write(json2csv(mappedTransactions, { header: false }));
+              res.write(`\n`);
+            } while (result.transactions.totalCount > result.transactions.limit + result.transactions.offset);
```
Comment (znarf) [inline src/server/controllers/account-transactions.js:368]: Pagination is here.

**Review by kewitz — APPROVED** [review]
Clever use of streaming here, I like it very much.
The code looks clean but I was not able to test this.

File: src/server/controllers/account-transactions.js (hunk 3 above)
Comment (kewitz) [reply src/server/controllers/account-transactions.js:296]: Cool, I got why and I think it makes sense.

**Pull request merged**